src/main/java/com/caiporalang/lexic/state/IdentifierState.java

```diff
@@ -0,0 +1,27 @@
+package com.caiporalang.lexic.state;
+
+import com.caiporalang.lexic.ScannerContext;
+import com.caiporalang.lexic.Token;
+import com.caiporalang.lexic.TokenType;
+import com.caiporalang.lexic.exception.InvalidIdentifier;
+
+import static com.caiporalang.lexic.state.States.createIdentifierState;
+
+class IdentifierState extends ScannerState {
+
+    public IdentifierState(ScannerContext context, String term) {
```

Comment:
There's no need for this constructor to be public. I suggest changing it to package private